kviz/visualizer.py

```diff
@@ -350,16 +405,32 @@ def fit(self, X, Y, snap_freq=10, filename='decision_boundary', duration=1000, *
         else:
             epochs = snap_freq
 
-        for _ in range(int(epochs / snap_freq)):
+        temp_dir = ".snapshots"
+        os.makedirs(temp_dir, exist_ok=True)
+
+        for epoch in range(int(epochs / snap_freq)):
             self.model.fit(X, Y, epochs=snap_freq, **kwargs)
             self._int_models = self._get_int_models()  # TODO: make this function more efficient
+            if (view_feature_space):
+                if (len(self.model.layers) > 3):
+                    raise ValueError("The model must have only one hidden layer for this visualization")
+                images.append(im.fromarray(self._snap_feature_space(X, Y, filename)))
+            else:
+                images.append(im.fromarray(self._snap_decision_boundary(X, Y, filename)))
```

Comment:
you can remove this block - only binary crossentropy and mean squared error losses are supported